packages/contract-utils/access-control/src/storage.rs

```diff
@@ -0,0 +1,492 @@
+///
+/// # Arguments
+///
+/// * `e` - Access to Soroban environment.
+/// * `caller` - The address of the caller, must be the admin or has the
+///   `AdminRole` privileges for this role.
+/// * `account` - The account to revoke the role from.
+/// * `role` - The role to revoke.
+///
+/// # Errors
+///
+/// * `AccessControlError::Unauthorized` - If the `caller` does not have enough
+///   privileges.
+/// * `AccessControlError::AccountNotFound` - If the `account` doesn't have the
+///   role.
+/// * refer to [`remove_from_role_enumeration()`] errors.
+///
+/// # Events
+///
+/// * topics - `["role_revoked", role: Symbol, account: Address]`
+/// * data - `[sender: Address]`
+pub fn revoke_role(e: &Env, caller: &Address, account: &Address, role: &Symbol) {
+    ensure_if_admin_or_admin_role(e, caller, role);
+
+    // Check if account has the role
+    if has_role(e, account, role).is_none() {
+        panic_with_error!(e, AccessControlError::AccountNotFound);
+    }
+
+    remove_from_role_enumeration(e, account, role);
+
+    let key = AccessControlStorageKey::HasRole(account.clone(), role.clone());
+    e.storage().persistent().remove(&key);
+
+    emit_role_revoked(e, role, account, caller);
+}
+
+/// Allows an account to renounce a role assigned to itself.
+/// Users can only renounce roles for their own account.
+///
+/// # Arguments
+///
+/// * `e` - Access to Soroban environment.
+/// * `caller` - The address of the caller, must be the account that has the
+///   role.
+/// * `role` - The role to renounce.
+///
+/// # Errors
+///
+/// * `AccessControlError::AccountNotFound` - If the `caller` doesn't have the
+///   role.
+/// * refer to [`remove_from_role_enumeration()`] errors.
+///
+/// # Events
+///
+/// * topics - `["role_revoked", role: Symbol, account: Address]`
+/// * data - `[sender: Address]`
+pub fn renounce_role(e: &Env, caller: &Address, role: &Symbol) {
+    if has_role(e, caller, role).is_none() {
+        panic_with_error!(e, AccessControlError::AccountNotFound);
+    }
+
+    remove_from_role_enumeration(e, caller, role);
+
+    let key = AccessControlStorageKey::HasRole(caller.clone(), role.clone());
+    e.storage().persistent().remove(&key);
+
+    emit_role_revoked(e, role, caller, caller);
+}
+
+/// Initiates admin role transfer.
+/// Admin privileges for the current admin are not revoked until the
+/// recipient accepts the transfer.
+/// Overrides the previous pending transfer if there is one.
+///
+/// # Arguments
+///
+/// * `e` - Access to Soroban environment.
+/// * `caller` - The address of the caller, must be the admin.
+/// * `new_admin` - The account to transfer the admin privileges to.
+///
+/// # Errors
+///
+/// * `AccessControlError::Unauthorized` - If the `caller` is not the admin.
+///
+/// # Events
+///
+/// * topics - `["admin_transfer_started", current_admin: Address]`
+/// * data - `[new_admin: Address]`
+pub fn transfer_admin_role(e: &Env, caller: &Address, new_admin: &Address) {
```

Comment:
why not using the same mechanism for setting/canceling the admin by using `live_until_ledger` as in [token approvals](https://github.com/OpenZeppelin/stellar-contracts/blob/35c921a22341337484252f32975f0dfde4c9d1ec/packages/tokens/non-fungible/src/storage.rs#L307)?